src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1161,6 +1164,106 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: DataSource,
     }
   }
 
+
+  def getWorkspaceStatus(workspaceName: WorkspaceName, userSubjectId: Option[String]): Future[PerRequestMessage] = {
+    asAdmin {
+      dataSource.inFutureTransaction(readLocks = Set(workspaceName)) { txn =>
+        withWorkspaceContext(workspaceName, txn) { workspaceContext =>
+          val STATUS_FOUND = "FOUND"
+          val STATUS_NOT_FOUND = "NOT_FOUND"
+          val STATUS_CAN_WRITE = "USER_CAN_WRITE"
+          val STATUS_CANNOT_WRITE = "USER_CANNOT_WRITE"
+          val STATUS_NA = "NOT_AVAILABLE"
+
+          val bucketName = workspaceContext.workspace.bucketName
+          val rawlsGroupRefs = workspaceContext.workspace.accessLevels
+          val googleGroupRefs = rawlsGroupRefs map { ref =>
+            ref._1 -> containerDAO.authDAO.loadGroup(ref._2, txn)
+          }
+
+          val userRef = userSubjectId match {
```

Comment:
you can do this with map instead of match

userSubjectId.map(id => containerDAO.authDAO.loadUser(RawlsUserRef(RawlsUserSubjectId(id)), txn))